crates/core/tedge_actors/src/runtime.rs

```diff
@@ -261,9 +261,12 @@ async fn run_task(
     task: Box<dyn Task>,
     running_name: String,
 ) -> Result<String, (String, RuntimeError)> {
-    task.run().await.map_err(|e| (running_name.clone(), e))?;
-
-    Ok(running_name)
+    match tokio::spawn(task.run()).await {
```

Comment:
What was the benefit in spawning the task first and awaiting on it immediately over calling `run()` directly? What was preventing the error mapping with the former direct `run()` call version?

Reply:
> returning the error directly with that ? was the problem, right?
No, this change doesn't concern the Result of run its about the run method panicking.

Originally, had the run method panicked the future would result in a `Err(JoinError)` without me having the chance do anything in run_task, the JoinError doesn't know our actors name, which meant that the runtime on line 220 where it handles the JoinError, doesn't know which actor has finished and can't clean up the running actors or send the appropriate `RuntimeEvent`.

By spawning another task here I can match on the result of that `JoinHandle` and if its a `Err(JoinError)` map it so I return the name of the actor.